### PR TITLE
Separate module for library. Parcelable models

### PR DESCRIPTION
**Pull request by d-kunin**
TLDR: this is copy of https://github.com/kaaes/spotify-web-api-android/pull/76 which was merged by mistake and reverted by me.

We have two big change here.
### Multi-module project
Now project has separate module 'library' with library sources and tests. We can add more modules like 'samples' and 'echonest' later.
Gradle files a bit more structured now:
1. _gradle.properties_ contains versions of all dependencies to use, and also current library version
2. _release.gradle_ hides javadoc, jacoco, and aar renaming
3. _build.gradle_ became pretty slim root project file with no boilerplate
4 _library/build.gradle_ describes library building and testing

### All models are Parcelable 
Pretty much everything in the package 'kaaes.spotify.webapi.android.models' is Parcelable now.
There is a bunch of tests in `kaaes.spotify.webapi.android.ParcelableModelsTest` to provide some confidence.

Sorry for almost 2k lines :)
ping @kaaes

### CODE REVIEW

**Comment by kaaes** [comment]
@d-kunin I think this looks great!
